java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java

```diff
@@ -3111,14 +3111,57 @@ public Void visitClass(ClassTree node, Void p) {
         return scanner.completesNormally;
     }
 
-    public static void performRewriteRuleSwitch(JavaFix.TransformationContext ctx, TreePath tp, Tree st) {
+    public static boolean isCompatibleWithSwitchExpression(SwitchTree st) {
+        boolean firstCase = true;
+        Name leftTreeName = null;
+        for (CaseTree ct : st.getCases()) {
+            List<StatementTree> statements = new ArrayList<>(ct.getStatements());
+            switch (statements.size()) {
```

Comment:
In case of more statements, we can create block with break value. This can be considered in separate enhancement.